### PR TITLE
ensure thread safety in clear octomap

### PR DESCRIPTION
**Pull request by simonschmeisser**
### Description
As commented in https://github.com/ros-planning/moveit/pull/2320#issuecomment-746096098 I recently encountered some crashes when clearOctomap is being called while some planners still run. Even though that is a weird thing to do, it still shouldn't crash.

Also #2320 uses `UPDATE_SCENE` which causes a full scene to be transmitted while `UPDATE_GEOMETRY` should be sufficient.

### Checklist
- [x] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [x] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

### CODE REVIEW

**Comment by simonschmeisser** [comment]
@felixvd friendly ping as you were the last one to touch this code

**Review by v4hn — APPROVED** [review]
Looks good to me. Awaits a second review.

File: moveit_ros/planning/planning_scene_monitor/src/planning_scene_monitor.cpp
Comment (felixvd) [inline moveit_ros/planning/planning_scene_monitor/src/planning_scene_monitor.cpp:536]: It wasn't hard to figure out how this works, but I had to think for a second because I don't use mutexes a lot. Do you want to add a comment to the effect of `// Lock scene using mutex before removing the octomap. Lock expires after the code block` for the less familiar? I'm on the fence.

**Review by felixvd — APPROVED** [review]
My bad or introducing the bug. LGTM. Thanks for the fix of UPDATE_GEOMETRY, too.

Will merge after you decide about the proposed comment. I'm good either way.

File: moveit_ros/planning/planning_scene_monitor/src/planning_scene_monitor.cpp
Comment (simonschmeisser) [reply moveit_ros/planning/planning_scene_monitor/src/planning_scene_monitor.cpp:536]: I should probably add a comment about the extra brackets that are there so that the scoped lock gets lifted before notifying consumers (`triggerSceneUpdateEvent`) in order to avoid a deadlock ...

Comment (felixvd) [reply moveit_ros/planning/planning_scene_monitor/src/planning_scene_monitor.cpp:536]: I didn't know that the lock needs to be lifted before that call. How about adding `// Lift the scoped lock before calling triggerSceneUpdateEvent to avoid deadlock` at the end of the block?

I'm fine with not adding the other comment because I suppose it's not "[nonobvious to a reader who understands C++ well]](https://google.github.io/styleguide/cppguide.html#Implementation_Comment_Donts)".

File: moveit_ros/planning/planning_scene_monitor/src/planning_scene_monitor.cpp
Comment (felixvd) [inline moveit_ros/planning/planning_scene_monitor/src/planning_scene_monitor.cpp:548]:
```suggestion
    }  // Lift the scoped lock before calling triggerSceneUpdateEvent to avoid deadlock
```

**Comment by v4hn** [comment]
Thanks @simonschmeisser . Congratulations on getting pull request number #2500. :clown_face: 
We sure deal with a lot of patches as time passes...

**Comment by felixvd** [comment]
I hope we don't have to wait until [#12345](https://bugs.python.org/issue12345) to merge [this](https://github.com/ros-planning/moveit_tutorials/pull/554)

**Comment by simonschmeisser** [comment]
It took me a long time to actually find something to fix just so I could get this PR number ... :wink: 

Thanks for reviewing and merging ... I suspect there's going to be another one as we had crashes in the octree->clear part as well that should be locked by themselves

**Pull request merged**
